### DISCUSSION TITLE
Module Command

### DISCUSSION THREAD

**Discussion by angelhdzmultimedia** [nuxtrdev/nuxtr-vscode#123] [Ideas]
Hello!

Is anyone working on Module Command?
Otherwise I'm willing to contribute it myself. 🤝

I already have a VSCode Snippet which adds a new module, so it's just a matter of providing that code and modifying the templates with the data from the message boxes/prompts.

```ts
import { defineNuxtModule, createResolver } from 'nuxt/kit'

export default defineNuxtModule({
  meta: {
    name: 'name',
    configKey: 'configKey',
    version: 'version'
  },

  setup() {
    const { resolve } = createResolver(import.meta.url)
  }
})
```

**Comment by adhamfarrag** [comment]
Hello @angelhdzmultimedia,

I thought about doing so but then felt modules rarely rely on a single file, and therefore I thought I can just support `Nuxtr: Create new Nuxt Project` which you can select a module from and locate it wherever you want. How often do you authorize modules and how often do they come in a single file ?

**Reply by angelhdzmultimedia** [reply]
I personally authorize modules in all my projects because I use a modular approach as in NestJS, so I have auth module, api module where I inject auth controller and its auth service as a $api useNuxtApp provided property, DI, etc. but I'm aware people are rarely doing this in Nuxt development. People instead use Pinia stores. 

So yeah, `Nuxtr: Create new Nuxt Project > Module` is perfect, Thank you for your time! 🔥🤩🤝
